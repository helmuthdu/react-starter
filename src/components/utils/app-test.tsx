import { RouterContext } from 'next/dist/shared/lib/router-context';
import { IntlProvider } from 'react-intl';
import { RecoilRoot } from 'recoil';

const router = jest.mock('next/router', () => {
  const mockedRouter = {
    back: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    prefetch: jest.fn(),
    route: '/mock-route',
    pathname: 'mock-path'
  };
  return {
    default: mockedRouter,
    useRouter: jest.fn(() => mockedRouter),
    withRouter: (component: any) => {
      component.defaultProps = {
        ...component.defaultProps,
        router: mockedRouter
      };
      return component;
    }
  };
});

type Props = { children: any };
export const AppTest = ({ children }: Props) => (
  <RouterContext.Provider value={router as any}>
    <RecoilRoot>
      <IntlProvider locale="en" onError={() => undefined}>
        {children}
      </IntlProvider>
    </RecoilRoot>
  </RouterContext.Provider>
);
