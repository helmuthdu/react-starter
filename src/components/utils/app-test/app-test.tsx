import React from 'react';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import { store } from '../../../stores';

export const AppTest = ({ children }: any) => (
  <IntlProvider locale="en" onError={jest.fn()}>
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  </IntlProvider>
);
