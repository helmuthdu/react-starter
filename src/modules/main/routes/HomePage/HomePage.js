// @flow
import { push } from 'connected-react-router';
import React, { Component } from 'react';
import { frontloadConnect } from 'react-frontload';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import logo from '../../assets/images/logo.svg';
import { AUTH_ROUTES } from '../../../auth/router';
import { toggleLoading } from '../../../../store/modules/ui';
import { isLoading } from '../../../../store/modules/ui/getters';
import { MAIN_ROUTES } from '../../router';
import { Button, Header, Intro, Logo, Title, Wrapper } from './HomePage.styled';

interface HomeProps {
  isLoading: boolean;
  toggleLoading: Function;
  changePage: Function;
}

interface HomeState {}

export class HomePage extends Component<HomeProps, HomeState> {
  componentDidMount() {
    this.props.toggleLoading();
  }

  render() {
    return (
      <Wrapper>
        <Header onClick={this.props.toggleLoading}>
          <Logo src={logo} alt="logo" isLoading={this.props.isLoading} />
          <Title>Welcome to React</Title>
        </Header>
        <Intro>
          To get started, edit <code>src/index.js</code> and save to reload.
        </Intro>
        Navigate to
        <Button onClick={() => this.props.changePage(MAIN_ROUTES.ABOUT)} title="go to about page">
          about page
        </Button>
        or to
        <Button onClick={() => this.props.changePage(AUTH_ROUTES.SIGN_IN)} title="go to sign-in page">
          sign-in page
        </Button>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: isLoading(state.ui)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      toggleLoading,
      changePage: path => push(path)
    },
    dispatch
  );

// Request initial data for the component
const beforeMount = async props =>
  await new Promise((resolve, reject) => {
    console.log('Before mounting HomePage component');
    return resolve();
  });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(frontloadConnect(beforeMount)(HomePage));
