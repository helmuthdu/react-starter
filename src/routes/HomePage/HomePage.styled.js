import styled, { keyframes } from 'styled-components';

export const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const Wrapper = styled.div`
  text-align: center;
`;

export const Header = styled.header`
  background-color: #222;
  height: 250px;
  padding: 20px;
  color: white;
`;

export const Intro = styled.p`
  font-size: large;
`;

export const Logo = styled.img`
  animation: ${props => (props.isLoading ? `${rotate360} infinite 20s linear` : `none`)};
  height: 80px;
`;

export const Title = styled.h1`
  font-size: 1.5em;
`;
