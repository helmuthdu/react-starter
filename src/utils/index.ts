export * from './list.utils';
export * from './search.util';
