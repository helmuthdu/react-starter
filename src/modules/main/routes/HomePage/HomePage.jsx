// @flow
import { push } from 'connected-react-router';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { toggleLoading } from '../../../../store/modules/ui';
import { isLoading } from '../../../../store/modules/ui/getters';
import { AUTH_ROUTES } from '../../../auth/router';
import logo from '../../assets/images/logo.svg';
import { MAIN_ROUTES } from '../../router';

import './HomePage.scss';

type Props = {
  isLoading: boolean,
  toggleLoading: () => void,
  linkTo: (path: string) => void
};

export class HomePage extends Component<Props> {
  async componentDidMount() {
    await new Promise((resolve, reject) => {
      console.log('Before mounting HomePage component', this.props);
      return resolve();
    });
    this.props.toggleLoading();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header" onClick={this.props.toggleLoading}>
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
            Learn React
          </a>
          Navigate to
          <div className="App-link" onClick={() => this.props.linkTo(MAIN_ROUTES.ABOUT)} title="go to about page">
            about page
          </div>
          or to
          <div className="App-link" onClick={() => this.props.linkTo(AUTH_ROUTES.SIGN_IN)} title="go to sign-in page">
            sign-in page
          </div>
        </header>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: isLoading(state.ui)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      toggleLoading,
      linkTo: path => push(path)
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomePage);
