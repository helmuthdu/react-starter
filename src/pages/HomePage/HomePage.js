// @flow
import { push } from 'connected-react-router';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import logo from '../../logo.svg';
import { toggleLoading } from '../../store/modules/ui';
import { isLoading } from '../../store/modules/ui/getters';
import { Header, Intro, Logo, Title, Wrapper } from './HomePage.styled';

interface HomePageProps {
  isLoading: boolean;
  toggleLoading: Function;
  changePage: Function;
}

interface HomePageState {}

export class HomePage extends Component<HomePageProps, HomePageState> {
  componentDidMount() {
    this.props.toggleLoading();
  }

  render() {
    return (
      <Wrapper>
        <Header onClick={this.props.toggleLoading}>
          <Logo src={logo} alt="logo" isLoading={this.props.isLoading} />
          <Title>Welcome to React</Title>
        </Header>
        <Intro>
          To get started, edit <code>src/index.js</code> and save to reload.
        </Intro>
        Navigate to
        <strong onClick={this.props.changePage} title="go to about page">
          about page
        </strong>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: isLoading(state.ui)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      toggleLoading,
      changePage: () => push('/about')
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomePage);
