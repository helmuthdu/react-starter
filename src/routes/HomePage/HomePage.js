// @flow
import { push } from 'connected-react-router';
import React, { Component } from 'react';
import { frontloadConnect } from 'react-frontload';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import logo from '../../assets/images/logo.svg';
import { toggleLoading } from '../../store/modules/ui';
import { isLoading } from '../../store/modules/ui/getters';
import { Header, Intro, Logo, Title, Wrapper } from './HomePage.styled';

interface HomeProps {
  isLoading: boolean;
  toggleLoading: Function;
  changePage: Function;
}

interface HomeState {}

export class HomePage extends Component<HomeProps, HomeState> {
  componentDidMount() {
    this.props.toggleLoading();
  }

  render() {
    return (
      <Wrapper>
        <Header onClick={this.props.toggleLoading}>
          <Logo src={logo} alt="logo" isLoading={this.props.isLoading} />
          <Title>Welcome to React</Title>
        </Header>
        <Intro>
          To get started, edit <code>src/index.js</code> and save to reload.
        </Intro>
        Navigate to
        <button onClick={this.props.changePage} title="go to about page">
          about page
        </button>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: isLoading(state.ui)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      toggleLoading,
      changePage: () => push('/about')
    },
    dispatch
  );

// Request initial data for the component
const asyncData = async props => await new Promise((resolve, reject) => resolve());

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(frontloadConnect(asyncData)(HomePage));
