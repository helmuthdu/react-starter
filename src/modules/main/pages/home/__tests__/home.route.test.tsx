import { render } from '@testing-library/react';
import { AppTest } from '@/components/utils/app-test';
import HomePage from '../home.page';

describe('Route -> Home', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <AppTest>
        <HomePage />
      </AppTest>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
