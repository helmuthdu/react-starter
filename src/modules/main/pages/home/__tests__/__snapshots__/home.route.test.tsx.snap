// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route -> Home should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <img
        alt="logo"
        src="/static/assets/images/logo.svg"
      />
      <p>
        Edit 
        <code>
          src/app.js
        </code>
         and save to reload.
      </p>
      <a
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      Navigate to
      <a
        href="/about"
        title="go to about page"
      >
        about page
      </a>
      or to
      <a
        href="/sign-in"
        title="go to sign-in page"
      >
        sign-in page
      </a>
    </header>
  </div>
</DocumentFragment>
`;
